src/coercions.rs

```diff
@@ -1,16 +1,17 @@
-use deltalake::{
-    Schema as DeltaSchema, SchemaDataType as DeltaDataType, SchemaField as DeltaField,
-};
+use deltalake::{Schema as DeltaSchema, SchemaDataType as DeltaDataType};
 
 use chrono::prelude::*;
-use serde_json::{Map, Number, Value};
+use serde_json::Value;
 use std::collections::HashMap;
 use std::str::FromStr;
 
 #[derive(Debug, Clone, PartialEq)]
+#[allow(unused)]
 enum CoercionNode {
     Coercion(Coercion),
     Tree(CoercionTree),
+    ArrayTree(CoercionTree),
+    ArrayRaw(Coercion),
```

Comment:
I like the distinction at this level. What do you think about `ArrayPrimitive` instead of `ArrayRaw` for the name?

Reply:
yeap, changed